### DISCUSSION TITLE
How do I import the css into my own css or scss file when including beer via npm?

### DISCUSSION THREAD

**Discussion by bobdobbs** [beercss/beercss#198] [Q&A]
I'd like to include beer in my own project, for which I'm importing dependancies via npm.
Lucky for me beer is available via npm. So I've installed beer using 'npm i --savedev beercss'.

I have the the file /projectroot/src/sass/main.scss.
This is the entry point for my scss.

I figure that because css is valid scss, I should be able to import beer files into my main.scss.
But I don't know how to do this.

**Comment by laurentpayot** [comment]
@bobdobbs I’m using PostCSS now (with Vite) but when I was using Node Sass in my main Sass file I was using[`@import`](https://sass-lang.com/documentation/at-rules/import/) like so:

```sass
@import node_modules/beercss/src/cdn/beer.css
```
(Actually I was not importing beer.css but selecting only the css files I needed)

**Reply by FunMiles** [reply]
Do you mind giving more details on how you select the files you need?

I tried to use `import 'beercss'` in a sveltekit project and the app barfed complaining about the import. I ended up putting a `@import` for the cdn in the base .css file, but would prefer something that might be smaller.

PS: Here is what doing the import directly gives. Should I report this as an issue?
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for beertest/node_modules/.pnpm/beercss@3.4.10_typescript@5.3.3/node_modules/beercss/dist/cdn/beer.min.css

```

**Reply by laurentpayot** [reply]
@FunMiles I have no idea how to import CSS files in a Sveltekit project. But if I’m right Svelte uses Vite as a build tool, so I guess [these docs](https://vitejs.dev/guide/features#css) are relevant and importing a `.css` file should work.
From the path in your error message I see that you’re using (the excellent) pnpm. I had some issues myself with Vite and pnpm. I had to create a `.npmrc` file at the root of my projects with `shamefully-hoist=true
` inside to make some modules work. Can you try this?

To pick parts of Beercss I simply create my own version of [beer.css](https://github.com/beercss/beercss/blob/main/src/cdn/beer.css) with import lines starting with `../../node_modules/beercss/src/cdn/` instead of `./`, and of course with commented lines for the imports I don’t want. Then I import this customized file normally in my main.css file.

Is it of any help?

**Reply by FunMiles** [reply]
Thanks, @laurentpayot, I'll have a look at the links you sent. I'm able to do an @import inside of the main `.css` file, so for now at least, I can work with that.
For the question on the parts you use, I guess I should have been a bit more detailed. Importing the full `.css` file is over 64kb, which is not that small. But the complexity of beercss structure doesn't make it easy to know what parts to import. I'm not sure vite can shake down the used part only. Do you have a systematic approach to figure out what you're actually using?

**Reply by laurentpayot** [reply]
@FunMiles Well I just know what parts of Beercss I _don’t_ need. For instance I’m styling my dialog backrounds with [`dialog::backdrop`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#css) so I don’t need elements/overlays.css. I’m using custom styles with [accent-color](https://web.dev/articles/accent-color) so I don’t need elements/selections.css nor elements/sliders.css.
I don’t import these elements but maybe that’s overkill because anyway I also remove all the unused rules with the [PurgeCSS](https://github.com/FullHuman/purgecss) PostCSS plugin in Vite. PurgeCSS really removes a lot of unused stuff.

**Reply by nickthiru** [reply]
> Do you mind giving more details on how you select the files you need?
> 
> I tried to use `import 'beercss'` in a sveltekit project and the app barfed complaining about the import. I ended up putting a `@import` for the cdn in the base .css file, but would prefer something that might be smaller.
> 
> PS: Here is what doing the import directly gives. Should I report this as an issue?
> 
> ```
> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for beertest/node_modules/.pnpm/beercss@3.4.10_typescript@5.3.3/node_modules/beercss/dist/cdn/beer.min.css
> ```

As of this date, for anyone facing this TypeError when trying to use Beer CSS with SvelteKit (in a general manner, and not necessarily selecting the files you need), I did the following...

After installing the `beercss `and `material-dynamic-colors` packages via `npm`, I created a top-level (next to app.html) `app.css` and included the following import statement:
```
@import 'beercss/dist/cdn/beer.min.css';
```

Then, in the (top-level) `/src/routes/+layout.svelte`, I included the following imports:
```
<script>
  import "material-dynamic-colors";
  import "../app.css";
</script>
```

Note: Since `material-dynamic-colors` provides a JavaScript file (material-dynamic-colors.min.js) and not a CSS file, you'll need to include it using JavaScript.
